lib/mongoid/errors/invalid_around_callback.rb

```diff
@@ -0,0 +1,16 @@
+# frozen_string_literal: true
+
+module Mongoid
+  module Errors
+    # This error is raised when an around callback is
+    # defined by the user without a yield
+    class InvalidAroundCallback < MongoidError
```

Comment:
As per comment elsewhere, I am strongly opposed to introducing a new runtime error which users will need to handle in their code.

Reply:
@johnnyshields -- this exception is raised only for a single case: *when an around callback fails to yield.* This will *always* indicate a bug in the users code. If we did not raise the exception here, it would manifest instead as a `FiberError`, which would be confusing to users because it does not describe what the issue is.

Rather than risk confusing users with a seeming non-sequitur, we're opting to raise this new error to help direct them to where the bug is in their code. If you always yield from your around callbacks, you'll never encounter the error.

Reply:
@jamis I understand that, but today (prior to the Fiber version) you can do an around callback which doesn't yield and it doesn't raise any error as far as I'm aware--it just doesn't continue the callback chain.

Is there a possibility that users are using non-yielding callbacks for flow control of their actions?

I think this should be looked at in more detail...

Reply:
Thanks for your concern about this issue, @johnnyshields, but we really have investigated this and and *have* looked at this, in quite a bit of detail.

An around callback that fails to yield is not as innocuous as you think. Not only does it just fail to "continue the callback chain" -- *it fails to perform the intended operation at all*. That is to say, if you have an `around_save` without a yield, and you save the record, it will silently fail to save the record at all.

This is why I said that this is a bug in their code, and I don't feel bad about surfacing this noisily now, where we didn't before. If someone's code "breaks" because of this, it will break in a way that it should have broken before. I do not believe there is any way that anyone was using non-yielding around callbacks for any useful purpose, because they wouldn't have been functional at all.

Reply:
Yes, what if someone is using that behavior like a validation in order to intentionally abort the save flow? For example:

```ruby
around_save :do_not_save_if_red

def do_not_save_if_red
  if color == "red"
    # report error somehow
  else
    yield
  end
end
```

App owners will now need to check that no callback is implementing the above pattern somehow.